app/controllers/exercises_controller.rb

```diff
@@ -37,6 +37,10 @@ def max_intervention_count_per_exercise
     1
   end
 
+  def teacher_defined_test?
```

Comment:
I was not sure what would be cleaner here.

Either you could do the check within one "big" line:
`@embed_options[:disable_score] = true unless @exercise.files.find {|f| f.role == 'teacher_defined_test' }`

Or, as I did, you could define a method to check if an exercise has a test and then use this method.

Reply:
I would rather add a model method to the `Exercise`. This could be reused in more cases. Here is a proposal:

```ruby
def teacher_defined_assessment?
  files.any?(&:teacher_defined_assessment?)
end
```